examples/go/charm/charmcraft.yaml

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: go-k8s
+
+base: ubuntu@24.04
+
+platforms:
+  amd64:
+
+summary: An example Go application.
+
+description: An example Go application.
+
+actions:
+  rotate-secret-key:
+    description: Rotate the secret key. Users will be forced to log in again.
+      This might be useful if a security breach occurs.
+
+assumes:
+  - k8s-api
+
+config:
+  options:
+    port:
+      type: int
+      description: Default port where the application will listen
+      default: 8080
+    metrics-port:
+      type: int
+      description: Port where the prometheus metrics will be scraped.
+      default: 8080
+    metrics-path:
+      type: string
+      description: Path where the prometheus metrics will be scraped.
+      default: /metrics
+    secret-key:
```

Comment:
I didn't find where this config is used in the application. Shouldn't it be used somewhere?

Reply:
It is used in here: https://github.com/canonical/paas-app-charmer/blob/1dde01c5849be17f621c4f12544e203882cabf84/paas_app_charmer/app.py#L127 

Clearly a bit hidden...

Reply:
I haven't used it in the examples, but it will always be as an environment variable to the charm (it will be in the charmcraft extension).